CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs

```diff
@@ -29,34 +29,63 @@ internal static class ImageLoaderHelper
         internal static Dictionary<string, string> SupportedImageFormats =
             new() { { "Supported formats", "*.jpg;*.jpeg;*.jfif;*.png;*.bmp;*.tiff;*.tif;*.webp" } };
 
+        #region Waifu2X
         private static Waifu2X _waifu2X;
+        private static Waifu2XStatus _cachedStatus = Waifu2XStatus.NotInitialized;
 
-        public static Waifu2XStatus Waifu2XStatus => _waifu2X?.Status ?? Waifu2XStatus.NotAvailable;
+        public static Waifu2XStatus Waifu2XStatus
+        {
+            get
+            {
+                // Cache the status of waifu2x
+                if (_cachedStatus == Waifu2XStatus.NotInitialized)
+                {
+                    using var test = CreateWaifu2X();
+                    {
+                        _cachedStatus = test.Status;
+                    }
+                }
+                return _cachedStatus;
+            }
+        }
 
         public static bool IsWaifu2XEnabled
         {
             get => GetAppConfigValue("EnableWaifu2X").ToBool() && IsWaifu2XUsable;
-            set => SetAndSaveConfigValue("EnableWaifu2X", value);
+            set
+            {
+                SetAndSaveConfigValue("EnableWaifu2X", value);
+                if (value) InitWaifu2X();
+                else DestroyWaifu2X();
+            }
         }
 
-        public static bool IsWaifu2XUsable => Waifu2XStatus != Waifu2XStatus.NotAvailable && Waifu2XStatus != Waifu2XStatus.TestNotPassed;
+        public static bool IsWaifu2XUsable => Waifu2XStatus < Waifu2XStatus.Error;
+
+        private static Waifu2X CreateWaifu2X()
+        {
+            var waifu2X = new Waifu2X();
+            if (waifu2X.Status < Waifu2XStatus.Error)
+            {
+                waifu2X.SetParam(Param.Noise, -1);
+                waifu2X.SetParam(Param.Scale, 2);
+                waifu2X.Load(Path.Combine(AppFolder!, @"Assets\Waifu2X_Models\scale2.0x_model.param.bin"),
+                    Path.Combine(AppFolder!, @"Assets\Waifu2X_Models\scale2.0x_model.bin"));
+            }
```

Comment:
Saving the status here to the "cache" variable seems to solve the crash.

Reply:
fixed in https://github.com/CollapseLauncher/Collapse/pull/428/commits/087d635bb67d16cd8ee99a91b56d245ea4b01e2e